src/main/java/org/opentripplanner/raptor/rangeraptor/multicriteria/CalculateTransferToDestination.java

```diff
@@ -37,14 +37,28 @@
   @Override
   public void notifyElementAccepted(ArrivalView<T> newElement) {
     if (newElement.arrivedByTransit()) {
-      for (RaptorAccessEgress egress : egressPaths) {
-        destinationArrivals.add(newElement, egress);
-      }
+      addOnBoardStopArrivalToDestination(newElement);
     } else if (newElement.arrivedByTransfer()) {
-      for (RaptorAccessEgress egress : egressPaths) {
-        if (egress.stopReachedOnBoard()) {
-          destinationArrivals.add(newElement, egress);
-        }
+      addOnStreetArrivalToDestination(newElement);
+    } else if (newElement.arrivedByAccess()) {
+      if (newElement.arrivedOnBoard()) {
+        addOnBoardStopArrivalToDestination(newElement);
+      } else {
+        addOnStreetArrivalToDestination(newElement);
+      }
+    }
```

Comment:
I need to check with @t2gran but I think nothing should happen in the else case.

Reply:
The `else if (newElement.arrivedByAccess())` is probably not necessary, but informative. I will fix it with a ` else { assertIsAccess() ...}` instead.

Reply:
It would be nice to use a switch in these cases, but we can not use sealed classes here. I will have a fast look and see if we can use Enums instead - interesting to see if it affect performance.

Reply:
I changed the "arrivedByXyz()" methods to be replaced by `arrivedBy() : PathLegType`. This allow us to enforce all cases handled in switch statements instead of `if`s - it also make the code a bit shorter. I would be supprised if this affect performance (either way), but I think using an enum here is nice - it also allow us to leave the "inheritance model" if we would like to do that later.

Reply:
Nice work!